src/Tests/Microsoft.NET.TestFramework/Attributes/WindowsOnlyRequiresMSBuildVersionFactAttribute.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.InteropServices;
+using Xunit;
+
+namespace Microsoft.NET.TestFramework
+{
+    public class WindowsOnlyRequiresMSBuildVersionFactAttribute : FactAttribute
+    {
+        public WindowsOnlyRequiresMSBuildVersionFactAttribute(string version)
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                this.Skip = "This test requires Windows to run";
+            }
+
+            if (TestContext.Current.ToolsetUnderTest.ShouldUseFullFrameworkMSBuild)
```

Comment:
This "requires MSBuild version" logic is now in 4 different attributes, right?  Let's factor it out into a static method they can all call.